File: insight-brain-model/src/main/java/com/sonatype/insight/brain/model/trending/Applications.java
/*
 * Copyright (c) 2011-2013 Sonatype, Inc. All rights reserved.
 * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
 * "Sonatype" is a trademark of Sonatype, Inc.
 */
package com.sonatype.insight.brain.model.trending;

import java.util.List;

public class Applications
{
  private int total;

  private List<ApplicationRiskSummary> risks;

  public Applications() {
  }

  public Applications(int total, List<ApplicationRiskSummary> risks) {
    this.total = total;
    this.risks = risks;
  }

  public int getTotal() {
    return total;
  }

  public List<ApplicationRiskSummary> getRisks() {
    return risks;
  }
}

File: insight-brain-frontend/src/main/frontend/dashboard/results/applications/applications.js
/*
 * Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
 * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
 * "Sonatype" is a trademark of Sonatype, Inc.
 */
import template from './applications.html';
import resultsControllerMixinFactory from '../resultsControllerMixinFactory';

export default {
  template,
  controller: applicationsController,
  controllerAs: 'vm'
};

const RESULTS_TYPE = 'applications';

function applicationsController($ngRedux, dashboardDataService, $scope, actions) {
  const vm = this;

  Object.assign(vm, resultsControllerMixinFactory($ngRedux, dashboardDataService, $scope, actions, RESULTS_TYPE), {
    encodeURIComponent: window.encodeURIComponent
  });
}

applicationsController.$inject = [
  '$ngRedux', 'dashboard.data.service', '$scope', 'dashboardResultsActions'
];
